agent/engine/docker_task_engine.go

```diff
@@ -1581,6 +1581,29 @@ func (engine *DockerTaskEngine) pullAndUpdateContainerReference(task *apitask.Ta
 			field.ImageDigest: imageDigest,
 			field.ImageRef:    imageRef,
 		})
+	} else if referenceutil.TagExists(imageRef) {
+		digest := referenceutil.GetDigestFromImageRef(imageRef)
+		if digest != "" {
```

Comment:
I think if `imageDigest` from L1556 is empty then this `digest` would be empty too since a digest from image reference is used to initialize `container.ImageDigest` during transition to MANIFEST_PULLED state.